src/tensorwaves/optimizer/minuit.py

```diff
@@ -16,27 +16,30 @@ class Minuit2(Optimizer):
     Implements the `~.interfaces.Optimizer` interface.
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(self, callbacks: Optional[Iterable[Callback]] = None) -> None:
+        self.__callbacks = CallbackList([])
+        if callbacks is not None:
+            self.__callbacks = CallbackList(callbacks)
 
     def optimize(self, estimator: Estimator, initial_parameters: dict) -> dict:
         parameters = initial_parameters
-        progress_bar = tqdm()
 
-        @tf_file_logging(iterations=2)
         def __call_estimator(params: dict) -> float:
             estimator.update_parameters(params)
-            progress_bar.update()
-            return estimator()
+            estimator_value = estimator()
+            self.__callbacks.call_all(
+                parameters=params,
+                estimator_value=estimator_value,
+            )
+            return estimator_value
 
-        def __func(pars: list) -> float:
-            """Wrap the estimator."""
+        def __wrapped_function(pars: list) -> float:
```

Comment:
I originally split out this `__call_estimator` function to work with this decorator pattern. Since we are using the callback, I would merge `__call_estimator` with `__wrapped_function` again.

Reply:
74bda11